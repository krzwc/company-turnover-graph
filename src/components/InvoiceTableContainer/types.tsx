export type Props = {
    client: string
}

export interface Data {
    invoicesByClient: Array<{ id: string; issueDate: string; net: number; }>;
};