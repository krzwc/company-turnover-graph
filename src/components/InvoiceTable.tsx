import {
    createStyles,
    Grid,
    Paper,
    Table,
    TableBody,
    TableCell,
    TableHead,
    TableRow,
    Theme,
    withStyles,
    WithStyles,
} from '@material-ui/core';
import { Link } from 'react-router-dom'
import { formatInPLN } from '../utils/formatInPLN'
import * as React from 'react';

const styles = (theme: Theme) =>
    createStyles({
        paper: {
            marginTop: theme.spacing.unit * 3,
            overflowX: 'auto',
        },
        table: {
            minWidth: '80vw'
        },
        link: {
            textDecoration: 'none',
            color: 'rgba(0, 0, 0, 0.54)',
            '&:visited': {
                color: 'rgba(0, 0, 0, 0.54)'
            }
        }
    });

interface Props extends WithStyles<typeof styles> {
    data: Array<{ id: string, issueDate: string, net: number}>
}

const InvoiceTable = (props: Props) => {
    const { classes, data } = props;
    const rows = data.map((inv, index) => {
        const id = index+1
        return { id: id, invoiceId: inv.id, issueDate: inv.issueDate, netAmount: inv.net}
    })

    return (
        <Grid
            container
            direction="column"
            alignItems="center"
        >
        <Grid item>
        <Paper className={classes.paper}>
            <Table className={classes.table} padding="dense">
                <TableHead>
                    <TableRow>
                        <TableCell><Link to="/" className={classes.link}>â‡¦ Go Back</Link></TableCell>
                        <TableCell />
                        <TableCell />
                    </TableRow>
                    <TableRow>
                        <TableCell>Invoice number</TableCell>
                        <TableCell align={'right'}>Date of issue</TableCell>
                        <TableCell align={'right'}>Net amount</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {rows.map(row => {
                        return (
                            <TableRow key={row.id}>
                                <TableCell component="th" scope="row">
                                    {row.invoiceId}
                                </TableCell>
                                <TableCell align={'right'}>{row.issueDate}</TableCell>
                                <TableCell align={'right'}>{formatInPLN(row.netAmount)}</TableCell>
                            </TableRow>
                        );
                    })}
                </TableBody>
            </Table>
        </Paper>
        </Grid>
        </Grid>
    );
};

export default withStyles(styles)(InvoiceTable);