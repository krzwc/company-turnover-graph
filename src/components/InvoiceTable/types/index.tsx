import {WithStyles} from "@material-ui/core";
import { styles } from "../styles"

export interface Props extends WithStyles<typeof styles> {
    data: Array<{ id: string, issueDate: string, net: number}>
}