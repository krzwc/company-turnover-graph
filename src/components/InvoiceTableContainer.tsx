import React from "react";
import { Query } from "react-apollo";
import gql from "graphql-tag";
import InvoiceTable from './InvoiceTable'

type Props = {
    client: string
}

interface Data {
    invoicesByClient: Array<{ id: string; issueDate: string; net: number; }>;
};

export default ({ client }: Props) => (
    <Query<Data> query={GET_INVOICES_BY_CLIENT} variables={{ client }}>
        {({ data, loading }) => {
            if (loading) return "Loading...";
            return (
                data &&
                data.invoicesByClient &&
                <InvoiceTable data={data.invoicesByClient} />
            );
        }}
    </Query>
);

export const GET_INVOICES_BY_CLIENT = gql`
    query invoicesByClient($client: String!) {
        invoicesByClient(client: $client) {
            id
            issueDate
            net
        }
    }
`;
