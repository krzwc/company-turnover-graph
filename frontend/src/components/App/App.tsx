import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';

import BubbleContainer from '../bubble-container';
import InvoiceTableContainer from '../invoice-table-container';
import './app.css';
import { Props } from './types';
import { endpoint } from '../../utils/endpoint';

const client = new ApolloClient({
    uri: endpoint,
});

const App: React.FC = () => {
    return (
        <ApolloProvider client={client}>
            <Router>
                <div className="App">
                    <Route exact path="/" component={BubbleContainer} />
                    <Route
                        path="/:client"
                        render={({ match }: Props) => <InvoiceTableContainer client={match.params.client} />}
                    />
                </div>
            </Router>
        </ApolloProvider>
    );
};

export default App;
