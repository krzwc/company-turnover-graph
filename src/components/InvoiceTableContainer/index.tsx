import InvoiceTableContainer from './InvoiceTableContainer';

export default InvoiceTableContainer;