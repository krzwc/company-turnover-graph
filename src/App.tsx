import React from 'react';
import { BrowserRouter as Router, Route, RouteComponentProps } from 'react-router-dom';
import BubbleContainer from './components/BubbleContainer'
import InvoiceTableContainer from './components/InvoiceTableContainer'
import ApolloClient from "apollo-boost";
import { ApolloProvider } from "react-apollo";
import './App.css';

const client = new ApolloClient({
    uri: "http://localhost:4000"
});

interface MatchParams {
    client: string;
}

interface Props extends RouteComponentProps<MatchParams> {
}

const App: React.FC = () => {
  return (
      <ApolloProvider client={client}>
          <Router>
            <div className="App">
                <Route exact path="/" component={ BubbleContainer } />
                <Route path='/:client' render={( {match}: Props) => (
                    <InvoiceTableContainer client={match.params.client} /> )} />
            </div>
          </Router>
      </ApolloProvider>
  );
}

export default App;
