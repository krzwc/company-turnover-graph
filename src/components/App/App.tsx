import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import BubbleContainer from '../BubbleContainer'
import InvoiceTableContainer from '../InvoiceTableContainer'
import ApolloClient from "apollo-boost";
import { ApolloProvider } from "react-apollo";
import './App.css';
import { Props } from "./types"
import { endpoint } from '../../utils/endpoint'

const client = new ApolloClient({
    uri: endpoint
});

const App: React.FC = () => {
  return (
      <ApolloProvider client={client}>
          <Router>
            <div className="App">
                <Route exact path="/" component={ BubbleContainer } />
                <Route path='/:client' render={( {match}: Props) => (
                    <InvoiceTableContainer client={match.params.client} /> )} />
            </div>
          </Router>
      </ApolloProvider>
  );
}

export default App;
