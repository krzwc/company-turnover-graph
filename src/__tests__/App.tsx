import React from 'react'
import { MemoryRouter, Route } from 'react-router-dom'
import * as enzyme from 'enzyme';
import App from '../App'
import InvoiceTableContainer from "../components/InvoiceTableContainer";
import Bubble from "../components/Bubble";
import InvoiceTable from '../components/InvoiceTable'
import { mockData, mockSize } from './Bubble'
import { tableMockData } from './InvoiceTable'

test('React Router snapshot matches', () => {
    const app = enzyme.shallow(
        <MemoryRouter initialEntries={['/']} initialIndex={0}>
                <App/>
            </MemoryRouter>);

    expect(app).toMatchSnapshot();
});

test('React Router "/"', () => {
    const app = enzyme.mount(
        <MemoryRouter initialEntries={['/']} initialIndex={0}>
            <Route exact path='/' component={ () => <Bubble data={mockData} size={mockSize} />} />
        </MemoryRouter>);
    expect(
        app.contains(<svg width={100} height={100} />)
    ).toBe(true);
});

test('React Router "/" doesn\'t route to "/:client"', () => {
    const app = enzyme.mount(
        <MemoryRouter initialEntries={['/']}>
            <Route path='/:client' component={() => <InvoiceTable data={tableMockData} />} />
        </MemoryRouter>);
    expect(
        app.contains("FAEXP/5/10/2018")
    ).not.toBe(true);
});

test('React Router "/:somestring" -> renders a component', () => {
    const app = enzyme.mount(
        <MemoryRouter initialEntries={['/somestring']}>
            <Route path='/:client' component={() => <InvoiceTable data={tableMockData} />} />
        </MemoryRouter>);
    expect(
        app.contains("FAEXP/5/10/2018")
    ).toBe(true);
});