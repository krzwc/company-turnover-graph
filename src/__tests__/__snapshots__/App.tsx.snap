// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble snapshot matches 1`] = `
<div>
  <svg
    height={100}
    width={100}
  />
</div>
`;

exports[`InvoiceTable snapshot matches 1`] = `
<InvoiceTable
  classes={
    Object {
      "link": "InvoiceTable-link-3",
      "paper": "InvoiceTable-paper-1",
      "table": "InvoiceTable-table-2",
    }
  }
  data={
    Array [
      Object {
        "id": "FAEXP/5/10/2018",
        "issueDate": "10/10/2018",
        "net": 5000,
      },
      Object {
        "id": "FAEXP/4/12/2018",
        "issueDate": "12/12/2018",
        "net": 6000,
      },
    ]
  }
/>
`;

exports[`React Router snapshot matches 1`] = `ShallowWrapper {}`;
